File: app/src/main/java/com/example/yue/nexttext/DataType/Time.java
package com.example.yue.nexttext.DataType;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.Calendar;

/**
 * Created by jamesmulvenna on 2017-09-28.
 */

public class Time implements Parcelable {
    private String date, time;

    public Time(String newDate, String newTime) {
        this.date = newDate;
        this.time = newTime;
    }

    private Time(Parcel in) {
        date = in.readString();
        time = in.readString();
    }

    public static final Creator<Time> CREATOR = new Creator<Time>() {
        @Override
        public Time createFromParcel(Parcel in) {
            return new Time(in);
        }

        @Override
        public Time[] newArray(int size) {
            return new Time[size];
        }
    };

    public String getDate() {
        return this.date;
    }

    public void setDate(String newDate) {
        this.date = newDate;
    }

    public String getTime() {
        return this.time;
    }

    public void setTime(String newTime) {
        this.time = newTime;
    }


    public String getTimer(){

        String AM = "AM";
        String parseAM[] = time.split(":");

        int hour = Integer.parseInt(parseAM[0]);
        int minute = Integer.parseInt(parseAM[1]);

        if(hour > 12){
            hour -= 12;
            AM = "PM";
        }
        return hour + ":" + minute + AM;
    }

    @Override
    public String toString() { return date + " " + getTimer(); }

    public Calendar getCalendar(){
        Calendar calendar = Calendar.getInstance();

        //get date
        String parseDate[] = date.split("/");

        int day = Integer.parseInt(parseDate[0]);
        int month = Integer.parseInt(parseDate[1])-1;
        int year = Integer.parseInt(parseDate[2]);

        int hour;
        int minute;

        String parseAM[] = time.split(":");

        hour = Integer.parseInt(parseAM[0]);
        minute = Integer.parseInt(parseAM[1]);

        calendar.set(Calendar.YEAR, year);
        calendar.set(Calendar.MONTH, month);
        calendar.set(Calendar.DAY_OF_MONTH, day);
        calendar.set(Calendar.HOUR_OF_DAY, hour);
        calendar.set(Calendar.MINUTE, minute);
        calendar.set(Calendar.SECOND, 0);

        System.out.println(calendar.getTime().toString());

        return calendar;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(date);
        parcel.writeString(time);
    }
}

File: app/src/main/java/com/example/yue/nexttext/Core/DataType/Time.kt
package com.example.yue.nexttext.Core.DataType

import android.icu.util.Calendar
import android.os.Parcel
import android.os.Parcelable
import com.example.yue.nexttext.UI.Utilities

class Time(val date : String,
           val time: String): Parcelable {

    fun getCalendar(): Calendar{
        val calendar = Calendar.getInstance()
        calendar.set(Calendar.YEAR, Utilities.reverseDateFormat_YEAR(this.date))
        calendar.set(Calendar.MONTH, Utilities.reverseDateFormat_MONTH(this.date))
        calendar.set(Calendar.DAY_OF_MONTH, Utilities.reverseDateFormat_DAY(this.date))
        calendar.set(Calendar.HOUR_OF_DAY, Utilities.reverseTimeFormat_HOUR(this.time))
        calendar.set(Calendar.MINUTE, Utilities.reverseTimeFormat_MINUTE(this.time))
        return calendar
    }

    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(date)
        parcel.writeString(time)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Time> {
        override fun createFromParcel(parcel: Parcel): Time {
            return Time(parcel)
        }

        override fun newArray(size: Int): Array<Time?> {
            return arrayOfNulls(size)
        }
    }
}
